File: codigos/1 - estrutura sequencial/1021 - Notas e Moedas/Main.cc
// Online C compiler to run C program online
#include <stdio.h>

int main() {
    double valor; 
    scanf("%lf", &valor);
    
    valor += 0.00001;
    
    double notas[] = {100, 50, 20, 10, 5, 2};
    double moedas[] = {1, .5, .25, .1, .05, .01};
    
    printf("NOTAS:\n");
    for (int i = 0; i < 6; i++){
        double nota = notas[i];
        int qntd = valor/nota;
        valor -= qntd*nota;
        printf("%d nota(s) de R$ %.2lf\n", qntd, nota);
    }
    
    printf("MOEDAS:\n");
    for (int i = 0; i < 6; i++){
        double moeda = moedas[i];
        int qntd = valor/moeda;
        if (i == 5) valor = moeda;
        else valor -= qntd*moeda;
        printf("%d moeda(s) de R$ %.2lf\n", qntd, moeda);
    }
}

File: codigos/1 - estrutura sequencial/1021 - Notas e Moedas/Main.clj
(def notas [100 50 20 10 5 2])
(def moedas [1 0.5 0.25 0.1 0.05 0.01])

(defn notas-op 
  ([valor]
   (notas-op valor [] 0))
  ([valor resultados i]
   (if (= i (count notas))
     [valor resultados]
     (let [nota (notas i)
           qntd (int (/ valor nota))]
       (recur (- valor (* qntd nota)) (conj resultados qntd) (inc i))))))

(defn moedas-op
  ([valor]
   (moedas-op valor [] 0))
  ([valor resultados i]
   (if (= i (count moedas))
     resultados
     (let [moeda (moedas i)
           qntd (int (/ valor moeda))]
       (recur (- valor (* qntd moeda)) (conj resultados qntd) (inc i))))))

(defn main []
  (let [valor (float (+ (read) 0.000001))
        [valor-apos-notas resultados-notas] (notas-op valor)
        resultados-moedas (moedas-op valor-apos-notas)]
    (println "NOTAS:")
    (doall (map #(println %2 "nota(s) de R$" (format "%.2f" (float %1))) notas resultados-notas))
    (println "MOEDAS:")
    (doall (map #(println %2 "moeda(s) de R$" (format "%.2f" (float %1))) moedas resultados-moedas))))

(main)